components/clp-py-utils/clp_py_utils/clp_metadata_db_utils.py

```diff
@@ -157,6 +157,30 @@ def fetch_existing_datasets(
     return {row["name"] for row in rows}
 
 
+def validate_and_cache_dataset(
```

Comment:
same comment, we might be able to avoid calling validate_and_cache_dataset when dataset is None.